[CSS-Sizing] Treat {min,max,fit}-content block length values as initial https://bugs.webkit.org/show_bug.cgi?id=264448 rdar://problem/118478408

Ahmad-S792 opened the pull request:
#### 4ccf66bbdf3b2aa920b197853f5eeaf9ed35638c
<pre>
[CSS-Sizing] Treat {min,max,fit}-content block length values as initial <a href="https://bugs.webkit.org/show_bug.cgi?id=264448">https://bugs.webkit.org/show_bug.cgi?id=264448</a> <a href="https://rdar.apple.com/problem/118478408">rdar://problem/118478408</a>

Reviewed by NOBODY (OOPS!).

Partial Merge: <a href="https://chromium.googlesource.com/chromium/src.git/+/3a0c4adda19bade345ca115fdcfe825acf0b8869">https://chromium.googlesource.com/chromium/src.git/+/3a0c4adda19bade345ca115fdcfe825acf0b8869</a>

The specification [1] says to treat the &apos;min-content&apos;, &apos;max-content&apos;
and &apos;fit-content&apos; values as the initial value of the property in question,
if said property is a block length property. In other words, these
CSS values really only make a difference in the inline direction.

[1] <a href="https://www.w3.org/TR/css-sizing-3/#sizing-values">https://www.w3.org/TR/css-sizing-3/#sizing-values</a>

* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::constrainLogicalHeightByMinMax const):
(WebCore::RenderBox::replacedMinMaxLogicalHeightComputesAsNone const):
(WebCore::RenderBox::computePositionedLogicalWidth const):
(WebCore::RenderBox::computePositionedLogicalHeight const):
* LayoutTests/TestExpectations: Progressed Tests
* LayoutTests/css3/flexbox/flexitem.html:
* LayoutTests/css3/flexbox/flexitem-expected.txt:
* LayoutTests/fast/css-intrinsic-dimensions/height-expected.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-positioned-expected.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-positioned-replaced-expected.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-positioned-replaced.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-positioned.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-replaced-expected.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-replaced.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height.html: Removed.
* LayoutTests/platform/ios/css3/flexbox/flexitem-expected.txt:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4ccf66bbdf3b2aa920b197853f5eeaf9ed35638c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75893 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/54922 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/28790 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80390 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27159 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/78009 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/64064 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3216 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/59506 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-position/position-absolute-fit-content.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17666 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78960 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49383 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65179 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39866 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/46783 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-position/position-absolute-fit-content.html (failure)") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/22662 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-position/position-absolute-fit-content.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/25486 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/67898 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23000 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-position/position-absolute-fit-content.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/81854 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3262 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/2060 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-position/position-absolute-fit-content.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/67737 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-position/position-absolute-fit-content.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3416 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65147 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67044 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10985 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/9113 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-position/position-absolute-fit-content.html (failure)") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3212 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/6018 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3233 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/4171 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/3240 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay reviewed (approved):
rs=me (I can't follow this logic anymore)

Ahmad-S792 committed:
[CSS-Sizing] Treat {min,max,fit}-content block length values as initial https://bugs.webkit.org/show_bug.cgi?id=264448 rdar://problem/118478408

Reviewed by NOBODY (OOPS!).

Partial Merge: https://chromium.googlesource.com/chromium/src.git/+/3a0c4adda19bade345ca115fdcfe825acf0b8869

The specification [1] says to treat the 'min-content', 'max-content'
and 'fit-content' values as the initial value of the property in question,
if said property is a block length property. In other words, these
CSS values really only make a difference in the inline direction.

[1] https://www.w3.org/TR/css-sizing-3/#sizing-values

* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::constrainLogicalHeightByMinMax const):
(WebCore::RenderBox::replacedMinMaxLogicalHeightComputesAsNone const):
(WebCore::RenderBox::computePositionedLogicalWidth const):
(WebCore::RenderBox::computePositionedLogicalHeight const):
* LayoutTests/TestExpectations: Progressed Tests
* LayoutTests/css3/flexbox/flexitem.html:
* LayoutTests/css3/flexbox/flexitem-expected.txt:
* LayoutTests/fast/css-intrinsic-dimensions/height-expected.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-positioned-expected.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-positioned-replaced-expected.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-positioned-replaced.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-positioned.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-replaced-expected.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height-replaced.html: Removed.
* LayoutTests/fast/css-intrinsic-dimensions/height.html: Removed.
* LayoutTests/platform/ios/css3/flexbox/flexitem-expected.txt:
